tobac/themes/tobac_v1/feature_detection.py

```diff
@@ -483,29 +561,97 @@ def filter_min_distance(features,dxy,min_distance):
+    for index_1, index_2 in indices:
         if index_1 is not index_2:
-            features.loc[index_1,'hdim_1']
-            distance=dxy*np.sqrt((features.loc[index_1,'hdim_1']-features.loc[index_2,'hdim_1'])**2+(features.loc[index_1,'hdim_2']-features.loc[index_2,'hdim_2'])**2)
+            features.loc[index_1, "hdim_1"]
+            distance = dxy * np.sqrt(
+                (features.loc[index_1, "hdim_1"] - features.loc[index_2, "hdim_1"]) ** 2
+                + (features.loc[index_1, "hdim_2"] - features.loc[index_2, "hdim_2"])
+                ** 2
+            )
             if distance <= min_distance:
-#                        logging.debug('distance<= min_distance: ' + str(distance))
-                if features.loc[index_1,'threshold_value']>features.loc[index_2,'threshold_value']:
+                #                        logging.debug('distance<= min_distance: ' + str(distance))
+                if (
+                    features.loc[index_1, "threshold_value"]
+                    > features.loc[index_2, "threshold_value"]
+                ):
                     remove_list_distance.append(index_2)
-                elif features.loc[index_1,'threshold_value']<features.loc[index_2,'threshold_value']:
+                elif (
+                    features.loc[index_1, "threshold_value"]
+                    < features.loc[index_2, "threshold_value"]
+                ):
                     remove_list_distance.append(index_1)
-                elif features.loc[index_1,'threshold_value']==features.loc[index_2,'threshold_value']:
-                    if features.loc[index_1,'num']>features.loc[index_2,'num']:
+                elif (
+                    features.loc[index_1, "threshold_value"]
+                    == features.loc[index_2, "threshold_value"]
+                ):
+                    if features.loc[index_1, "num"] > features.loc[index_2, "num"]:
                         remove_list_distance.append(index_2)
-                    elif features.loc[index_1,'num']<features.loc[index_2,'num']:
+                    elif features.loc[index_1, "num"] < features.loc[index_2, "num"]:
                         remove_list_distance.append(index_1)
-                    elif features.loc[index_1,'num']==features.loc[index_2,'num']:
+                    elif features.loc[index_1, "num"] == features.loc[index_2, "num"]:
                         remove_list_distance.append(index_2)
-    features=features[~features.index.isin(remove_list_distance)]
+    features = features[~features.index.isin(remove_list_distance)]
     return features
+
+
+def spectral_filtering(dxy, field_in, lambda_min, lambda_max):
+    """
+    This function creates and applies a 2D transfer function that can be used as a bandpass filter to remove
+    certain wavelengths of an atmospheric field (e.g. vorticity).
+
+    Args:
```

Comment:
Yes, this would be required! We specified this aspect in our CONTRIBUTING.md file. Unfortunately, this file is a bit hidden in the v2.0-dev branch ....

Reply:
Ah perfect! I completely missed that file, but thank you, it is very clear.